modules/vaos/app/services/vaos/messages_service.rb

```diff
@@ -18,14 +18,12 @@ def post_message(request_id, request_object_body)
         params = VAOS::MessageForm.new(user, request_id, request_object_body).params
         response = perform(:post, messages_url(request_id), params, headers)
 
-        if response.status == 200
-          {
-            data: OpenStruct.new(response.body),
-            meta: {}
-          }
-        else
-          handle_error(response)
-        end
+        handle_error(response) unless response.status == 200
+
+        {
+          data: OpenStruct.new(response.body.except(:sender_id)),
+          meta: {}
+        }
```

Comment:
I think this change introduces some coupling with the `handle_error` method. It assumes that `handle_error` will exit the code execution (which it currently does do by raising), but that assumption might not always be true.